### ISSUE TITLE
Deploying CKF to EKS tests fail waiting oidc-gatekeeper to go to Blocked

### ISSUE THREAD

**Issue by orfeas-k** [canonical/bundle-kubeflow#669]
### Problem
The scheduled Deploy CKF to EKS workflow started failing last week. Looking into it, it looks like `oidc-gatekeeper` never goes to `Blocked` as the [tests expect](https://github.com/canonical/bundle-kubeflow/blob/6adb65e85924015f246c06c52779b99df2c3bc14/tests/integration/test_bundle_deployment.py#L32-L38) and thus they timeout
```
asyncio.exceptions.TimeoutError: Timed out waiting for model:
oidc-gatekeeper/0 [idle] active:
```

### Reproduce
Manually run [the tests](https://github.com/canonical/bundle-kubeflow/actions/workflows/deploy-eks.yaml). 

### Cause
The tests deploy [`1.7/edge` bundle](https://github.com/canonical/bundle-kubeflow/blob/9f3917440a1601af6975526baa9ca4bdde64fff7/tests/integration/test_bundle_deployment.py#L11-L22) while [OIDC gatekeeper operator](https://charmhub.io/oidc-gatekeeper) uses [revision 250](https://github.com/canonical/oidc-gatekeeper-operator/releases/tag/rev250) for `1.7/edge`. Taking a look at rev250 code, we see that the [charm code](https://github.com/canonical/oidc-gatekeeper-operator/blob/rev250/src/charm.py) has the [_check_public_url](https://github.com/canonical/oidc-gatekeeper-operator/blob/rev250/src/charm.py#L133) method, but it seems like it is not called anywhere, unlike main that [implements](https://github.com/canonical/oidc-gatekeeper-operator/blob/main/src/charm.py#L139) and [calls](https://github.com/canonical/oidc-gatekeeper-operator/blob/main/src/charm.py#L57) the method. Thus, it is expected for `oidc-gatekeeper` to not go to `Blocked`. 

That being said, it looks like this wasn't always the case since the scheduled tests had been [passing for about a month](https://github.com/canonical/bundle-kubeflow/actions/workflows/deploy-eks.yaml).

### Solution
WRT to the tests, they should be modified to expect what is there in the charm's code. 
WRT to oidc-gatekeeper, I think we should investigate if/when/how we introduced a change to the charm, since we can see both in the [first](https://github.com/canonical/bundle-kubeflow/actions/runs/5582053427/job/15117079335#step:10:530) and the [last](https://github.com/canonical/bundle-kubeflow/actions/runs/5791333003/job/15695986642#step:10:589) successful runs of the tests, that `oidc-gatekeeper` was going to
```
INFO     juju.model:model.py:2597 Waiting for model:
  oidc-gatekeeper/0 [idle] blocked: public-url config required
```

**Comment by orfeas-k** [comment]
Looking at `oidc-gatekeeper` charm's [releases](https://github.com/canonical/oidc-gatekeeper-operator/releases), we see that 
- Revision 205: Released to `ckf-1.7/edge` at 21:38 UTC on 23 May 2023 which [has this line](https://github.com/canonical/oidc-gatekeeper-operator/blob/rev205/src/charm.py#L56) that checks public url and is actually a rewritten sidecar charm
- Revision 250: Released to `ckf-1.7/edge` at 14:12 UTC on **9 Aug 2023** (with Full Changelog: rev170...rev250) which is a PodSpec charm [without the check_public_url](https://github.com/canonical/oidc-gatekeeper-operator/blob/rev250/src/charm.py#L37) implemented.

This explains the above as the last successful run of the workflow was in August 8. 

In addition, the above tests will not have to change since we will implement checking for public url in `track/ckf-1.7` branch.

**Comment by orfeas-k** [comment]
Closing since tests pass after https://github.com/canonical/oidc-gatekeeper-operator/pull/108 was merged.
